### PR TITLE
#535 Bugfix for reduxStateBug and filtering

### PR DESCRIPTION
**Pull request by Mathriok**
In this pull request I've mainly tried to adress:
- When reduxState had a string in filterInput it would crash.
- The memory consumption when filtering

If we now have a string in filterInput-key, filteredLogs will be fetched in the first render instead of crashing. To battle too high memory consumption filteredLines gets fetched every 1,5 seconds in a running file (this can of course be changed to any other suitable delay). Along with the fetch comes previousFilteredLinesLength. The first time we fetch previousFilteredLinesLength is 0 and therefore we need to read the whole file. All the other times we only read the last 10 lines and filter those.  For this I've brought back and reused readNLines which was removed during the scrollback feature implementation i think.  Finally I've also added a dispatch function to clear the filtered logs when filterInput gets emptied. 

In https://github.com/kits-ab/LogLady/issues/543 I explain how I noticed that lines going to mainScriptOffLoader were not being sent one by one when wanted because of previousLinesLength.current being assigned an undefined value and therefore the slice in newLines doesn't slice at all. When altering this and sending one by one I ran into the crash and thought that this maybe is a seperate issue. For that reason those lines are commented. However I think these comments can be uncommented, and my key in sendMessageToHiddenWindow could be deleted when this has been fixed.

See how this works for you!

### CODE REVIEW

File: src/js/view/reducers/logViewerReducer.js [hunk 1]
```diff
@@ -183,16 +185,54 @@ export const logViewerReducer = (state = initialState, action) => {
...
-      const { sourcePath, filteredLines, lineCount } = action.data;
+      const { sourcePath, filteredLines, filterString } = action.data;
+      const previousLines = state.filteredLogs[sourcePath]
+        ? state.filteredLogs[sourcePath]
+        : [];
+      const previousFilterString = state.filterString[sourcePath]
+        ? state.filterString[sourcePath]
+        : '';
+
+      console.log({ previousLines, previousFilterString });
```
Comment (MelindaSW) [inline src/js/view/reducers/logViewerReducer.js]: Nice, just tried it out and so far so good. Found a stray console.log here that could perhaps be removed. Will try it some more and look at the code.

File: src/js/view/components/LogViewer.js [hunk 2]
```diff
@@ -129,63 +166,57 @@ const LogViewer = props => {
...
-    }
-
-    if (logs) {
-      // Reset the previous lines count, as all lines should be wiped.
-      previousFilteredLinesLength.current = 0;
-      previousLinesLength.current = 0;
+      // let newLines = filteredLogs.slice(previousFilteredLinesLength.current);
       sendMessageToHiddenWindow({
-        logs
+        // sendLinesOneByOne: previousFilteredLinesLength.current > 0 ? true : false,
```
Comment (MelindaSW) [inline src/js/view/components/LogViewer.js:186]: I am not sure if leaving these lines commented will make sense for future developers. We understand it when reading the pr and the bug ticket. Maybe remove the comments and save them as a note in the bug ticket you wrote?  Or ad an explanatory comment? Not sure atm what would be the best solution. I might need to read the explanation again and come back.

Comment (Atmanea) [reply src/js/view/components/LogViewer.js:186]: Yes, makes sense to at least leave an explanatory comment together with the code to hint why this is still here and refer it to ticket #543 .

Comment (MelindaSW) [reply src/js/view/components/LogViewer.js:186]: Other than that everything seems to work well on my machine. I have not been able to recreate the bug and the filtering process does not cost as much cpu as before. Thumbs up!

Comment (Mathriok) [reply src/js/view/components/LogViewer.js:189]: Good idea! I added explanatory comments that I hope will make it clearer and refered to #543

File: src/js/view/components/TopPanel.js [hunk 3]
```diff
@@ -54,7 +54,7 @@ class TopPanel extends React.Component {
               label={'Filter'}
               placeholder={
                 this.props.logSizes[sourcePath] > 10000000
-                  ? 'File too large'
+                  ? 'File too big'
```
Comment (MelindaSW) [inline src/js/view/components/TopPanel.js:57]: :)

**Pull request merged**